rpc_util_test.go

```diff
@@ -290,3 +292,170 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
 }
+
+// testCompressor is a mock implementation of an encoding.Compressor interface
+// used for testing compression and decompression functionality.
+// Compress is a placeholder for the compression logic.
+// It currently returns a nil WriteCloser and no error, as the actual compression is not implemented.
+type testCompressor struct {
+	decompressedData []byte
+	errDecompress    error
+	customReader     io.Reader
+}
+
+func (m *testCompressor) Compress(w io.Writer) (io.WriteCloser, error) {
+	return nil, nil
+}
+
+func (m *testCompressor) Decompress(r io.Reader) (io.Reader, error) {
+	if m.errDecompress != nil {
+		return nil, m.errDecompress
+	}
+	if m.customReader != nil {
+		return m.customReader, nil
+	}
+	return bytes.NewReader(m.decompressedData), nil
+}
+
+func (m *testCompressor) Name() string {
+	return "testCompressor"
+}
+
+type ErrorReader struct{}
+
+func (e *ErrorReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New("simulated io.Copy read error")
+}
+func TestDecompress(t *testing.T) {
+	bestCompressor, err := NewGZIPCompressorWithLevel(gzip.BestCompression)
+	if err != nil {
+		t.Fatalf("Could not initialize gzip compressor with best compression.")
+	}
+	bestSpeedCompressor, err := NewGZIPCompressorWithLevel(gzip.BestSpeed)
+	if err != nil {
+		t.Fatalf("Could not initialize gzip compressor with best speed compression.")
+	}
+
+	defaultCompressor, err := NewGZIPCompressorWithLevel(gzip.BestSpeed)
+	if err != nil {
+		t.Fatalf("Could not initialize gzip compressor with default compression.")
+	}
+
+	level5, err := NewGZIPCompressorWithLevel(5)
+	if err != nil {
+		t.Fatalf("Could not initialize gzip compressor with level 5 compression.")
+	}
+
+	for _, test := range []struct {
+		// input
+		data []byte
+		cp   Compressor
+		dc   Decompressor
+
+		// outputs
+		err error
+	}{
+		{make([]byte, 1024), NewGZIPCompressor(), NewGZIPDecompressor(), nil},
+		{make([]byte, 1024), bestCompressor, NewGZIPDecompressor(), nil},
+		{make([]byte, 1024), bestSpeedCompressor, NewGZIPDecompressor(), nil},
+		{make([]byte, 1024), defaultCompressor, NewGZIPDecompressor(), nil},
+		{make([]byte, 1024), level5, NewGZIPDecompressor(), nil},
+	} {
+		b := new(bytes.Buffer)
+		if err := test.cp.Do(b, test.data); err != test.err {
+			t.Fatalf("Compressor.Do(_, %v) = %v, want %v", test.data, err, test.err)
+		}
+		if b.Len() >= len(test.data) {
+			t.Fatalf("The compressor fails to compress data.")
+		}
+		if p, err := test.dc.Do(b); err != nil || !bytes.Equal(test.data, p) {
+			t.Fatalf("Decompressor.Do(%v) = %v, %v, want %v, <nil>", b, p, err, test.data)
+		}
+	}
+	tests := []struct {
+		name                  string
+		compressor            encoding.Compressor
+		input                 mem.BufferSlice
+		maxReceiveMessageSize int
+		want                  mem.BufferSlice
+		compressedsize        int
+		error                 error
+	}{
+		{
+			name: "Successful decompression",
+			compressor: &testCompressor{
```

Comment:
you have to use real compressor here. I have mentioned this so many times. Why do you keep using test compressor here?